hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerQueueManager.java

```diff
@@ -509,6 +509,12 @@ public AbstractLeafQueue createQueue(QueuePath queue)
     String leafQueueName = queue.getLeafName();
     String parentQueueName = queue.getParent();
 
+    if (StringUtils.isEmpty(leafQueueName)) {
+      throw new SchedulerDynamicEditException(
+              "Trying to create new queue=" + queue.getFullPath()
+                      + ", which has empty leaf shortname.");
+    }
+
```

Comment:
I don't think this part is necessary. QueueManager.createQueue(QueuePath) is only called from CapacityScheduler.getOrCreateQueueFromPlacementContext (at least from production code), and it has the following check right before this call:

```
  //we need to make sure there are no empty path parts present
  if (queuePath.hasEmptyPart()) {
    LOG.error("Application submitted to invalid path due to empty parts: " +
        "'{}'", queuePath);
    return null;
  }
```

So essentially we should not be able to reach this from production code, only from tests.

Reply:
Agreed with this. This path currently does fail due to the fact that `QueuePath.hasEmptyParts()` does not actually return true for an empty leaf name, but the added unit test for `QueueManager.createQueue(QueuePath)` is not necessary, so I will remove that in the next commit.